src/ota/mod.rs

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::error::DeviceManagerError;
+use crate::ota::rauc::BundleInfo;
+use async_trait::async_trait;
+use log::info;
+use mockall::automock;
+use zbus::export::futures_util::StreamExt;
+
+pub(crate) mod ota_handler;
+pub(crate) mod rauc;
+use crate::ota::rauc::RaucProxy;
+
+#[automock]
+#[async_trait]
+pub trait OTA: Send + Sync {
+    async fn install_bundle(&self, source: &str) -> Result<(), DeviceManagerError>;
+    async fn last_error(&self) -> Result<String, DeviceManagerError>;
+    async fn info(&self, bundle: &str) -> Result<BundleInfo, DeviceManagerError>;
+    async fn operation(&self) -> Result<String, DeviceManagerError>;
+    async fn compatible(&self) -> Result<String, DeviceManagerError>;
+    async fn boot_slot(&self) -> Result<String, DeviceManagerError>;
+    async fn receive_completed(&self) -> Result<i32, DeviceManagerError>;
+}
+
+pub struct OTARauc<'a> {
+    rauc: RaucProxy<'a>,
+}
+
+#[async_trait]
+impl<'a> OTA for OTARauc<'a> {
+    async fn install_bundle(&self, _source: &str) -> Result<(), DeviceManagerError> {
+        self.rauc
+            .install_bundle("path", std::collections::HashMap::new())
+            .await?;
+        Ok(())
+    }
```

Comment:
Let's try to fix the Codecov configuration because reviewing PRs interspersed with all these warnings is a little painful